### PR TITLE
Add manifest linting job to prow as a new CI check

### PR DESCRIPTION
**Pull request by fmuyassarov**
This adds a new manifestlint job to the prow configuration so the
manifestlint.sh script gets triggered for every pull request in
BMO, CAPM3, IPAM, HWCC repos. The task of the manifestlint.sh is
to validate only core types based manifests within those repos and
report if they don’t match Kubernetes OpenAPI specification.

### CODE REVIEW

**Comment by fmuyassarov** [comment]
/hold until below PRs get merged

- https://github.com/metal3-io/baremetal-operator/pull/733
- https://github.com/metal3-io/cluster-api-provider-metal3/pull/149
- https://github.com/metal3-io/ip-address-manager/pull/20
- https://github.com/metal3-io/hardware-classification-controller/pull/55

**Comment by kashifest** [comment]
looks good

**Review by maelk — APPROVED** [review]

**Comment by dhellmann** [comment]
I don't know if I have permission, but let's see

/lgtm

**Comment by fmuyassarov** [comment]
> I don't know if I have permission, but let's see
> 
> /lgtm

Seems no, unfortunately :(

**Pull request merged**
